src/view/socket.ts

```diff
@@ -2,30 +2,32 @@ import { Emitter } from '../core/emitter';
 import { EventsTypes } from '../events';
 import { IO } from '../io';
 import { Node } from '../node';
+import { NodeView } from './node';
+import { getOffset } from './utils';
 
 export class SocketView extends Emitter<EventsTypes> {
 
     el: HTMLElement;
     type: string;
     io: IO;
     node: Node;
+    nodeViewEl: HTMLElement;
 
-    constructor(el: HTMLElement, type: string, io: IO, node: Node, emitter: Emitter<EventsTypes>) {
+    constructor(el: HTMLElement, type: string, io: IO, node: Node, emitter: NodeView) {
```

Comment:
one small thing is the remaining `NodeView`. The element can be passed to `getPosition`

Reply:
@Ni55aN updated